llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -883,6 +881,110 @@ static bool removeDeviceGlobalFromCompilerUsed(Module &M) {
   return true;
 }
 
+// Add extra red zone to each image scope device globals if the module has been
+// instrumented by sanitizer pass. And record their infomation like size, red
+// zone size, beginning address.
+static bool instrumentDeviceGlobal(Module &M) {
+  auto &DL = M.getDataLayout();
+  IRBuilder<> IRB(M.getContext());
+  SmallVector<GlobalVariable *, 8> GlobalsToRemove;
+  SmallVector<GlobalVariable *, 8> NewDeviceGlobals;
+  SmallVector<Constant *, 8> DeviceGlobalMetadata;
+
+  constexpr uint64_t MaxRZ = 1 << 18;
+  const uint64_t MinRZ = 32;
+
+  Type *IntptrTy =
```

Comment:
```suggestion
  Type *IntTy =
```

Or maybe `SizeTTy`. That type is not a pointer, so `ptr` is definitely confusing here

Reply:
Changed to IntTy